File: frontend/src/types/Document.ts
type UserInfo = { id: number; username: string; role?: string };

export type Document = {
  id: number;
  title: string;
  content: string;
  status: "DRAFT" | "PENDING" | "APPROVED" | "REJECTED";
  createdBy: UserInfo;
  approvedBy?: UserInfo;
  approvedAt?: string;
  rejectedBy?: UserInfo;
  rejectedAt?: string;
};

export type CreateDocumentInput = {
  title: string;
  content: string;
};

File: backend/src/main/java/com/approval/model/Document.java
package com.approval.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "documents")
@Getter
@Setter
public class Document {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String title;

    private String content;

    @Enumerated(EnumType.STRING)
    private DocumentStatus status; // DRAFT, PENDING, APPROVED, REJECTED

    @ManyToOne
    @JoinColumn(name = "created_by_id", nullable = false)
    private User createdBy;

    @ManyToOne
    @JoinColumn(name = "approved_by_id")
    private User approvedBy;

    private LocalDateTime approvedAt;

    @ManyToOne
    @JoinColumn(name = "rejected_by_id")
    private User rejectedBy;

    private LocalDateTime rejectedAt;

    @OneToMany(mappedBy = "document", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<AuditLog> auditLogs = new ArrayList<>();
}
